Implement is_sym? method

akaryu0206 opened the pull request:
# New Features
- Implement `is_sym?` method (#36)
- Add test for `is_sym?` method

akaryu0206 committed:
[Add] Implement `is_sym?` method

 
- Implement `is_sym?` method
- Add test for `is_sym?` method

jagaapple commented on spec/string_foundation/symbol_spec.rb:
```diff
@@ -0,0 +1,48 @@
+    end
+
+    context "when an object equal to an argument," do
+      context "object and argument is only alphabets," do
```
Good context! If you could it at all, I'd like you to write tests to check whether a string includes only numbers and a string is an empty string.

jagaapple commented on spec/string_foundation/symbol_spec.rb:
```diff
@@ -0,0 +1,48 @@
+        it { is_expected.to be true }
+      end
+
+      context "object and argument is includes underscores," do
```
`is includes` -> `include`

jagaapple commented on spec/string_foundation/symbol_spec.rb:
```diff
@@ -0,0 +1,48 @@
+      end
+    end
+
+    context "when an object not equal to an argument," do
```
`not equal to` -> `is not equal to`

jagaapple commented on lib/string_foundation/symbol.rb:
```diff
@@ -0,0 +1,13 @@
+  def is_sym?(symbol)
+    raise ArgumentError.new("argument must be Symbol") unless symbol.instance_of?(Symbol)
+
+    self == symbol.to_s
```
Enclose a returning conditional value in parentheses because of be more readable. Read more [this](https://github.com/brushdown/string_foundation.rb/pull/32#discussion_r145046187).

```rb
(self == symbol.to_s)
```

akaryu0206 replied:
Sorry, I forgot...

jagaapple commented on lib/string_foundation/symbol.rb:
```diff
@@ -0,0 +1,13 @@
+# ==============================================================================
+# LIB - STRING FOUNDATION - SYMBOL
```
Change the name from The Symbol Methods to The Is Methods, so rename this to `is.rb` .

jagaapple reviewed (changes requested):
It was about 95% correct! Please modify very little things.

akaryu0206 committed:
[Clean] Improve reviewed codes

- Change the name for The Symbol Methods
- Add some tests to The Symbol Methods
- Fix wrong english
- Code beautify

Commit:
[Clean] Modify English for tests

- Modify English for tests

Commit:
Merge branch 'release/v1.1.0' of https://github.com/jagaapple/string_foundation into feature/implement-is-sym-method
